src/Application/Actions/UpdatePaymentMethod.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace MatchBot\Application\Actions;
+
+use Fig\Http\Message\StatusCodeInterface;
+use JetBrains\PhpStorm\Pure;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Psr\Log\LoggerInterface;
+use Stripe\Exception\InvalidRequestException;
+use Stripe\StripeClient;
+
+class UpdatePaymentMethod extends Action
+{
+    #[Pure]
+    public function __construct(
+        private StripeClient $stripeClient,
+        LoggerInterface      $logger
+    )
+    {
+        parent::__construct($logger);
+    }
+
+    protected function action(Request $request, Response $response, array $args): Response
+    {
+        $customerId = $request->getAttribute('pspId');
+        \assert(is_string($customerId));
+
+        $paymentMethodId = $args['payment_method_id'];
+        \assert(is_string($paymentMethodId));
+
+        /** @var array<array{id: string}> $allPaymentMethodsOfCustomer */
+        $allPaymentMethodsOfCustomer = $this->stripeClient->customers->allPaymentMethods(
+            $customerId,
+        )->toArray()['data'];
+
+        $allCustomersPaymentMethodIds = array_map(
+            static fn(array $pm): string => $pm['id'],
+            $allPaymentMethodsOfCustomer
+        );
```

Comment:
Quite a bit of this is duplicating what the DeletePaymentMethod controller does - not sure where to move it to to avoid duplication. Maybe we can live with the duplication for now, and then think about adding a service layer or something to the architecture before making a third duplicate.